metaflow/datatools/s3.py

```diff
@@ -527,14 +783,28 @@ def put_many(self, key_objs, overwrite=True):
 
         Args:
             key_objs:  (required) an iterator of (key, value) tuples. Value must
-                       be a string, bytes, or a unicode object.
+                       be a string, bytes, or a unicode object. Instead of
+                       (key, value) tuples, you can also pass any object that
+                       has the following properties 'key', 'value', 'content_type',
+                       'metadata' like the S3PutObject for example. 'key' and
+                       'value' are required but others are optional.
             overwrite: (optional) overwrites the key with obj, if it exists
 
         Returns:
             a list of (key, S3 URL) tuples corresponding to the files sent.
         """
         def _store():
-            for key, obj in key_objs:
+            for key_obj in key_objs:
+                key = getattr(key_obj, 'key', key_obj[0])
```

Comment:
you can't do `key_obj[0]` on an arbitrary object. Instead of `getattr`, just check `isinstance(key_obj, tuple)` and access `.key` and `value` directly in non-tuple objects

Reply:
Done